digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/pointasset/masstransitstop/TerminatedBusStopStrategy.scala

```diff
@@ -0,0 +1,54 @@
+package fi.liikennevirasto.digiroad2.service.pointasset.masstransitstop
+
+import java.util.NoSuchElementException
+
+import fi.liikennevirasto.digiroad2.{DigiroadEventBus, FloatingReason, Point}
+import fi.liikennevirasto.digiroad2.asset.{AdministrativeClass, Position, SimpleProperty}
+import fi.liikennevirasto.digiroad2.dao.{AssetPropertyConfiguration, MassTransitStopDao}
+import fi.liikennevirasto.digiroad2.linearasset.RoadLink
+import fi.liikennevirasto.digiroad2.service.RoadLinkService
+import fi.liikennevirasto.digiroad2.service.pointasset.masstransitstop.MassTransitStopOperations._
+import fi.liikennevirasto.digiroad2.util.GeometryTransform
+
+class TerminatedBusStopStrategy(typeId: Int, massTransitStopDao: MassTransitStopDao, roadLinkService: RoadLinkService, eventbus: DigiroadEventBus, geometryTransform: GeometryTransform) extends BusStopStrategy(typeId, massTransitStopDao, roadLinkService, eventbus, geometryTransform) {
+  val defaultChanges: Seq[String] = Seq(InventoryDateId, RoadName_FI, RoadName_SE, MassTransitStopTypePublicId)
+  val allowedChanges: Seq[String] = Seq(EndDate, AdministratorInfoPublicId)
+
+  def allowedPropertyChanges(properties: Set[SimpleProperty]): Boolean = {
+    properties.map(_.publicId).filterNot(defaultChanges.contains).forall(allowedChanges.contains)
+  }
+  def changesOnDefaultValues(properties: Set[SimpleProperty], asset: PersistedMassTransitStop): Boolean = {
+    properties.filter(prop => defaultChanges.contains(prop.publicId)).map(_.values).toSeq.diff(asset.propertyData.filter(prop => defaultChanges.contains(prop.publicId)).map(_.values)).nonEmpty
+  }
+
+  override def create(asset: NewMassTransitStop, username: String, point: Point, roadLink: RoadLink): (PersistedMassTransitStop, AbstractPublishInfo) = {
+    throw new UnsupportedOperationException
+  }
+  override def pickRoadLink(optRoadLink: Option[RoadLink], optHistoric: Option[RoadLink]): RoadLink = {
+    optHistoric.getOrElse(throw new NoSuchElementException)
+  }
+  override def is(newProperties: Set[SimpleProperty], roadLink: Option[RoadLink], existingAssetOption: Option[PersistedMassTransitStop], oldLink: Option[RoadLink]): Boolean = {
+    val properties = existingAssetOption match {
+      case Some(existingAsset) =>
+        (existingAsset.propertyData.
+          filterNot(property => newProperties.exists(_.publicId == property.publicId)).
+          map(property => SimpleProperty(property.publicId, property.values)) ++ newProperties).
+          filterNot(property => AssetPropertyConfiguration.commonAssetProperties.exists(_._1 == property.publicId))
+      case _ => newProperties.toSeq
+    }
+    val isOnTerminatedRoad = properties.find(_.publicId == MassTransitStopOperations.FloatingReasonPublicId).exists(_.values.headOption.exists(_.propertyValue == FloatingReason.TerminatedRoad.value.toString))
+    oldLink.nonEmpty && roadLink.isEmpty && isOnTerminatedRoad
+  }
```

Comment:
These condition (oldLink.nonEmpty && roadLink.isEmpty) don't make sense to check here, could give you a wrong 'is' information